Readme.md

```diff
@@ -0,0 +1,369 @@
+# Pls Run some jobs
+
+# Problem
+
+Running arbitrary linux processes on remote machine.
+
+# Approach
+
+## Library
+
+Individual commands are started as child processes. Handle is owned by `Job`, which streams logs of the process to a file and collects exit code. As a consequence, in the absence of cleanup policy _chatty_ processes would quickly exhaust available disk space. 
+
+Jobs are grouped together and owned by `Controller`, which serves as a container for jobs. Controller, deals with various aspects of spawning and responds to queries. 
+
+## Cgroups
+
+Library leverages cgroups to enforce resource limits. V2 is chosen for its simpler design (single hierarchy). Since controllers could be part of only one hierarchy and systemd mounts some controllers to v1 by default, `cgroup_no_v1=all` kernel boot param is used to opt-out of this behaviour and use unified (v2) hierarchy instead. See [playbook](./playbook.yml) for implementation details.
+
+Following controls are supported:
+- `cpu.weight`: provide value in [1, 100000] to set weight for spawned process (default 100), relative to other client processes.
+- `memory.high`: provide value in bytes to cap memory, process using more than this value will be throttled.
+- `memory.max`: provide value in bytes to hard-cap memory, process using more than this value will be OomKilled. 
+- `io.max.rbps` & `io.max.wbps`: provide value to cap io in bytes per second. 
+
+### Running as root
+
+Adding process to a cgroup requires eid of the process performing the move to be the owner of common ancestor of source and destination cgroups. In other words, for child C of process P belonging to non-root user U, to be moved in a cgroup owned by U, P itself needs to be in a cgroup that is direct ancestor of target cgroup and owned by U.   
+This could be achieved by setting up appropriate permissions beforehand as privileged user and executing binary as U.
+For this implementation it is a no-goal and process is ran as root. 
```

Comment:
Could you expand on your process execution model? It's not very clear from this description how the server will launch the jobs and place them into cgroups.

Reply:
Added **Flow** section below, pls lmk if I'm missing something and/or any further clarifications are needed. 
Do mermaid diagrams open for you? If for some reason they do not - code could be pasted @[mermaid live editor](https://mermaid.live/)